In [21]:
# importer les modules nécessaires
from bs4 import BeautifulSoup
import requests
# définir le lien à web scraper
url = 'https://openclassrooms.com/fr/courses/4462426-maitrisez-les-bases-de-donnees-nosql/4474691-etudiez-le-fonctionnement-d-elasticsearch'
# récupérer le contenu du site web
reponse = requests.get(url)
contenu = reponse.text
# parser le contenu HTML à l'aide de BeautifulSoup
soup = BeautifulSoup(contenu, features="html.parser")
# extraire le titre du site web
title = soup.title.string
print(title)
# extraire le premier paragraphe du contenu HTML
paragraph = soup.findAll('p')[4].text
print(paragraph)


                            Etudiez le fonctionnement d’Elasticsearch - Maîtrisez les bases de données NoSQL - OpenClassrooms
                    

course.header.alt.is_certifying



#Excercice 1
1. Le site web https://books.toscrape.com contient un catalogue de bouquins.
Vous allez récupérer la page web situé à l’endroit : Home > Books >
Classics et vous allez afficher le titre et le prix des bouquins (Classiques)
du moins cher au plus cher.

In [22]:
import requests
from bs4 import BeautifulSoup

# URL de la page contenant les classiques
url = "https://books.toscrape.com/catalogue/category/books/classics_6/index.html"

# Faire une requête GET à l'URL
response = requests.get(url)

# Vérifier si la requête a réussi
if response.status_code == 200:
    # Analyser le contenu HTML de la page
    soup = BeautifulSoup(response.content, 'html.parser')

    # Trouver tous les éléments correspondant aux livres classiques
    books = soup.find_all('li', class_='col-xs-6 col-sm-4 col-md-3 col-lg-3')

    # Créer une liste pour stocker les titres et prix des livres
    books_info = []

    # Parcourir chaque livre
    for book in books:
        # Extraire le titre du livre
        title = book.h3.a['title']

        # Extraire le prix du livre
        price = book.find('p', class_='price_color').text

        # Ajouter le titre et le prix à la liste des informations sur les livres
        books_info.append((title, price))

    # Trier les livres par prix (du moins cher au plus cher)
    sorted_books_info = sorted(books_info, key=lambda x: float(x[1].strip('£')))

    # Afficher les titres et prix des livres triés
    for book_info in sorted_books_info:
        print(f"Titre: {book_info[0]} - Prix: {book_info[1]}")
else:
    print("Erreur lors de la récupération de la page.")


Titre: The Hound of the Baskervilles (Sherlock Holmes #5) - Prix: £14.82
Titre: The Secret Garden - Prix: £15.08
Titre: Wuthering Heights - Prix: £17.73
Titre: The Complete Stories and Poems (The Works of Edgar Allan Poe [Cameo Edition]) - Prix: £26.78
Titre: Little Women (Little Women #1) - Prix: £28.07
Titre: The Metamorphosis - Prix: £28.58
Titre: The Picture of Dorian Gray - Prix: £29.70
Titre: Gone with the Wind - Prix: £32.49
Titre: Emma - Prix: £32.93
Titre: And Then There Were None - Prix: £35.01
Titre: Sense and Sensibility - Prix: £37.46
Titre: Beowulf - Prix: £38.35
Titre: The Story of Hong Gildong - Prix: £43.19
Titre: The Little Prince - Prix: £45.42
Titre: Of Mice and Men - Prix: £47.11
Titre: The Pilgrim's Progress - Prix: £50.26
Titre: Alice in Wonderland (Alice's Adventures in Wonderland #1) - Prix: £55.53
Titre: Animal Farm - Prix: £57.22
Titre: Candide - Prix: £58.63


#Excercice 2
Vous allez modifier votre bot de telle manière qu’il cherche un bouquin par
titre. Puis, affiche les informations associées avec bouquin, incluant le prix.
Par exemple, si l’utilisateur saisit "The Little Prince" , le prix "£45.42"
sera affiché

In [25]:
import requests
from bs4 import BeautifulSoup
import urllib.parse

def search_book_by_title(title):
    # Encode the search query
    encoded_title = urllib.parse.quote_plus(title)

    # Construct the search URL with the encoded query
    search_url = f"https://books.toscrape.com/catalogue/category/books_1/?q={encoded_title}"

    # Send a GET request to the search URL
    response = requests.get(search_url)

    # Check if the request was successful
    if response.status_code == 200:
        # Parse the HTML content of the search results page
        soup = BeautifulSoup(response.content, 'html.parser')

        # Find all books in the search results
        books = soup.find_all('article', class_='product_pod')

        # Check if any books were found
        if books:
            # Iterate over each book
            for book in books:
                # Extract book information
                book_title = book.h3.a['title']
                book_price = book.find('p', class_='price_color').text.strip()

                # Check if the title of the book matches the search query
                if title.lower() in book_title.lower():
                    # Print book information
                    print(f"Titre: {book_title} - Prix: {book_price}")
                    return  # Stop searching after finding the first matching book
            # If no book matches the search query
            print("Aucun livre trouvé avec ce titre.")
        else:
            print("Aucun livre trouvé avec ce titre.")
    else:
        print("Erreur lors de la recherche.")

# Demander à l'utilisateur de saisir le titre du livre
user_input = input("Entrez le titre du livre à rechercher: ")

# Rechercher le livre par titre et afficher ses informations
search_book_by_title(user_input)

Entrez le titre du livre à rechercher: Set Me Free
Titre: Set Me Free - Prix: £17.46
